File: backend/app/schemas/asset.py
from datetime import datetime

from pydantic import BaseModel


class ServiceOut(BaseModel):
    id: int
    port: int
    protocol: str
    service_name: str | None
    banner: str | None
    detected_at: datetime

    model_config = {"from_attributes": True}


class AssetOut(BaseModel):
    id: int
    ip_address: str
    hostname: str | None
    os_guess: str | None
    status: str
    first_seen: datetime
    last_seen: datetime

    model_config = {"from_attributes": True}


class AssetDetailOut(AssetOut):
    services: list[ServiceOut] = []

File: frontend/src/types/asset.ts
export interface Service {
  id: number;
  port: number;
  protocol: string;
  service_name: string | null;
  banner: string | null;
  detected_at: string;
}

export interface Asset {
  id: number;
  ip_address: string;
  hostname: string | null;
  os_guess: string | null;
  status: "up" | "down";
  first_seen: string;
  last_seen: string;
}

export interface AssetDetail extends Asset {
  services: Service[];
}
